lesson11/index.js

```diff
@@ -0,0 +1,82 @@
+const ul = document.getElementById('js-list');
+
+const createLoading = () => {
+    const li = document.createElement("li");
+    const loadingImg = document.createElement('img');
+
+    li.style.listStyle = "none";
+    li.id = "js-loading";
+    loadingImg.src = 'loading-circle.gif';
+    ul.appendChild(li).appendChild(loadingImg);
+};
+
+const removeLoading = () => document.getElementById('js-loading').remove();
+
+const createErrorMessage = (error) => {
+    const li = document.createElement("li");
+    li.textContent =`エラー内容：${error.message}`;
+    ul.appendChild(li);
+    console.error(error.message);
+};
+
+/**
+ * Async関数 
+ * @return {Array} 
+ */
+const jsonUrl = 'https://myjson.dit.upm.es/api/bins/4tpn';
+const fetchData = async () => {
+    try {
+        const response = await fetch(jsonUrl) // 中身はresponseオブジェクトを含むpromise
+        const json = await response.json(); //response オブジェクトから JSON を抽出
+        return json.data;
+
+    } catch(error) {
+        console.error(error);
+    } 
+};
+
+/**
+ * Async関数
+ * fetchData関数から値が返ってきた後、その値を返す
+ */
+const fetchListData = async () => {
+    createLoading();
+    try {
+        const data = await fetchData();
+        if( data.length === 0 ) {
+            throw new Error('データが空です。');
+        }
+        return data;
+
+    } catch(error) {
+        createErrorMessage(error);
+
+    }  finally {
+        removeLoading();
+    }
+};
+
+/**
+ * createList関数
+ * domのul内にli,aタグを作成し、fetchListDataで渡ってきた値を展開する
+ */
+const createList = async() => {
+    const val = await fetchListData();
+    const fragment = document.createDocumentFragment();
+    for (let i = 0; i < val.length; i++) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+    
+        a.href = `${val[i].a}.html`;
+        a.textContent = `${val[i].text}`;
+        img.src = val[i].img;
+        img.alt = val[i].alt;
+    
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+
+    ul.appendChild(fragment);
+};
```

Comment:
I thought the points I reviewed were OK.

I am very sorry, but I forgot to tell you one of the points reviewed by Monaka.
It is as follows.

> the values in "throw new Error("Data is empty");" are undefined, so it would be smoother to execute the code only when the values contain truthy values. The following example shows how to do this.
[Reference site](https://qiita.com/Zone_1997/items/225851ccdb921f253793)

```suggestion
const createList = async() => {
    const val = await fetchListData();
    const fragment = document.createDocumentFragment();
    //throw new Error("データが空です");のときのvalはundefinedなので
    if (val) { //valにtruthyな値が入っている時のみに実行させる
        for (let i = 0; i < val.length; i++) {
            const li = document.createElement('li');
            const a = document.createElement('a');
            const img = document.createElement('img');
    
            a.href = `${val[i].a}.html`;
            a.textContent = `${val[i].text}`;
            img.src = val[i].img;
            img.alt = val[i].alt;
    
            fragment.appendChild(li).appendChild(a).appendChild(img);
        }
        ul.appendChild(fragment);
    }
};
```

Reply:
Thank you for sharing.
I have fixed the code, please check it out.
[5e21bcf](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/11/commits/5e21bcf5010ef27664f9fa8a8b6863110261de08)

Reply:
Thank you for the many corrections, I will Approve!

nits
The indentation of the closing brackets "}" in the same place was off, 
so if you could fix it later, that would be great!

Reply:
I'm sorry for the careless mistake.
It has been corrected.
[0c9907d](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/11/commits/0c9907d1e9c3ede3d5f39429c2cee84c4d2b435e)